### DISCUSSION TITLE
RuntimeError with group license in docker

### DISCUSSION THREAD

**Discussion by Sandmann6d** [dashingsoft/pyarmor#1609] [Q&A]
Hello,
we are having problems using group license in docker, using a docker host for pyarmor-auth

Output of `pyarmor -d reg /path/to/pyarmor-device-regfile-xxxx.1.zip` looks correct:
![pyarmor reg](https://github.com/dashingsoft/pyarmor/assets/97108287/6c31860f-e440-49ad-98be-27977c2f1bbe)

Output of `pyarmor -d gen ...` contains the RuntimeError. Also showing content of `pyarmor.error.log`.
![pyarmor gen](https://github.com/dashingsoft/pyarmor/assets/97108287/ab2e4743-3093-4dcc-aced-0c96013446c9)

The docker host with `pyarmor-auth --debug` shows this (for `pyarmor -d reg` and `pyarmor -d gen` from the docker)
![pyarmor auth](https://github.com/dashingsoft/pyarmor/assets/97108287/c37ef3b9-a276-4f68-921b-ae0e64b5b264)

We tried pyarmor 8.4.3 and 8.4.4, pyarmor.cli.core 5.4.1 and 5.4.2.
How can we fix this? Thank you 🙏

**Comment by jondy** [comment]
I can see nothing about log and error message, please provide full log of each command.

**Reply by Sandmann6d** [reply]
I believe I have shared the full logs, minus identifying information.
Everywhere marked "machine id" is the same id.

Screenshots show:

1. pyarmor -d reg /path/to/pyarmor-device-regfile-xxxx.1.zip (docker)
2. pyarmor -d gen ... + pyarmor.error.log (lines 254-270) (docker)
3. pyarmor-auth --debug (docker host)

what information do you need?

**Reply by Sandmann6d** [reply]
hi :)
I'm not sure why, but adding `--network=host` in the build pipeline seems to have fixed it. So the docker command is `docker buildx build --network=host --add-host=host.docker.internal:172.17.0.1 ...`
Thank you
